platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -814,8 +1038,123 @@ namespace platoon_strategic_ihp
         }
     }
 
+    // UCLA: Mobility operation callback for lead_with_operation state (cut-in join).
+    void PlatoonStrategicIHPPlugin::mob_op_cb_leadwithoperation(const cav_msgs::MobilityOperation& msg)
+    {
+        std::string strategyParams = msg.strategy_params;
+        bool isPlatoonStatusMsg = (strategyParams.rfind(OPERATION_STATUS_TYPE, 0) == 0);
+        if(isPlatoonStatusMsg) 
+        {
+            mob_op_cb_STATUS(msg);
+            ROS_DEBUG_STREAM("Platoon STATUS operation message processed in Lead with Operation state.");
+        } 
+        else 
+        {
+            ROS_DEBUG_STREAM("Received a mobility operation message with params " << msg.strategy_params << " but ignored.");
+        }
+    }
+
+    // UCLA: Mobility operation callback for prepare to join state (cut-in join).
+    void PlatoonStrategicIHPPlugin::mob_op_cb_preparetojoin(const cav_msgs::MobilityOperation& msg)
+    {
+        /*
+         * If same lane with leader, then send request to do same lane join. 
+         * Otherwise, just send status params.
+         * Note: leader in state 'leading with operation' also send out INFO msg
+         */
+
+        // read parameters
+        std::string strategyParams = msg.strategy_params;
+        std::string senderId = msg.header.sender_id; 
+        std::string platoonId = msg.header.plan_id;  
+        std::string newLeaderID = msg.header.sender_id; 
+        std::string newLeaderPlatoonId = pm_.currentPlatoonID; 
```

Comment:
It's not clear why the platoon ID would be changing when we are doing a middle cut-in join.  I suggest a comment block explaining it.